### PR TITLE
Bump CS Fixer rules to follow latest Nextcloud standards

### PR DESCRIPTION
**Pull request by Altahrim**
## Summary
Update PHP CS coding standards to lastest Nextcloud standards

### CODE REVIEW

**Review by come-nc — COMMENTED** [review]
I think you should also change platform to php: 8.0 in the same composer.json file

File: core/Migrations/Version13000Date20170718121200.php [hunk 1]
```diff
@@ -888,25 +888,25 @@ public function changeSchema(IOutput $output, \Closure $schemaClosure, array $op
...
+		//			$table->addColumn('identifier', 'string', [
+		//				'notnull' => true,
+		//				'length' => 64,
+		//			]);
+		//			$table->addColumn('credentials', 'text', [
+		//				'notnull' => false,
+		//			]);
+		//			$table->setPrimaryKey(['user', 'identifier']);
+		//			$table->addIndex(['user'], 'credentials_user');
+		//		}
```
Comment (come-nc) [inline core/Migrations/Version13000Date20170718121200.php:909]: Just remove the dead code, no?

File: core/Migrations/Version16000Date20190207141427.php [hunk 2]
```diff
@@ -76,7 +76,7 @@ public function changeSchema(IOutput $output, Closure $schemaClosure, array $opt
 			]);
 
 			$table->setPrimaryKey(['collection_id', 'resource_type', 'resource_id'], 'crr_pk');
-//			$table->addUniqueIndex(['collection_id', 'resource_type', 'resource_id'], 'collres_unique_res');
+			//			$table->addUniqueIndex(['collection_id', 'resource_type', 'resource_id'], 'collres_unique_res');
```
Comment (come-nc) [inline core/Migrations/Version16000Date20190207141427.php:79]: Same

File: core/Migrations/Version17000Date20190514105811.php [hunk 3]
```diff
@@ -67,7 +67,7 @@ public function changeSchema(IOutput $output, Closure $schemaClosure, array $opt
 				'default' => 0,
 			]);
 			$table->setPrimaryKey(['fileid'], 'fce_pk');
-//			$table->addUniqueIndex(['fileid'], 'fce_fileid_idx');
+			//			$table->addUniqueIndex(['fileid'], 'fce_fileid_idx');
```
Comment (come-nc) [inline core/Migrations/Version17000Date20190514105811.php:70]: Same

File: core/Migrations/Version20000Date20201109081915.php [hunk 4]
```diff
@@ -43,7 +43,7 @@ public function changeSchema(IOutput $output, Closure $schemaClosure, array $opt
 
 		$result = $this->ensureColumnIsNullable($schema, 'share', 'password_by_talk');
 		$result = $this->ensureColumnIsNullable($schema, 'share', 'hide_download') || $result;
-//		$result = $this->ensureColumnIsNullable($schema, 'credentials', 'user') || $result;
+		//		$result = $this->ensureColumnIsNullable($schema, 'credentials', 'user') || $result;
```
Comment (come-nc) [inline core/Migrations/Version20000Date20201109081915.php:46]: Same

File: core/Migrations/Version13000Date20170718121200.php [hunk 5]
```diff
@@ -751,7 +751,6 @@ public function changeSchema(IOutput $output, \Closure $schemaClosure, array $op
 				'unsigned' => true,
 			]);
 			$table->setPrimaryKey(['objecttype', 'objectid', 'systemtagid'], 'som_pk');
-//			$table->addUniqueIndex(['objecttype', 'objectid', 'systemtagid'], 'mapping');
```
Comment (nickvergessen) [inline core/Migrations/Version13000Date20170718121200.php:754]: Please keep all the comments in migration files.
They are useful search results.
When dealing with old database schemas it's very useful to be able to quickly search if something is custom, or we modified/dropped it in a later version, etc.

File: core/Migrations/Version13000Date20170718121200.php [hunk 6]
```diff
@@ -888,26 +887,8 @@ public function changeSchema(IOutput $output, \Closure $schemaClosure, array $op
...
-//				'notnull' => true,
-//				'length' => 64,
-//			]);
-//			$table->addColumn('credentials', 'text', [
-//				'notnull' => false,
-//			]);
-//			$table->setPrimaryKey(['user', 'identifier']);
-//			$table->addIndex(['user'], 'credentials_user');
-//		}
-
```
Comment (nickvergessen) [inline core/Migrations/Version13000Date20170718121200.php:910]: Keep

File: core/Migrations/Version17000Date20190514105811.php [hunk 7]
```diff
@@ -67,7 +67,6 @@ public function changeSchema(IOutput $output, Closure $schemaClosure, array $opt
 				'default' => 0,
 			]);
 			$table->setPrimaryKey(['fileid'], 'fce_pk');
-//			$table->addUniqueIndex(['fileid'], 'fce_fileid_idx');
```
Comment (nickvergessen) [inline core/Migrations/Version17000Date20190514105811.php:70]: Keep

File: core/Migrations/Version20000Date20201109081915.php [hunk 8]
```diff
@@ -43,7 +43,6 @@ public function changeSchema(IOutput $output, Closure $schemaClosure, array $opt
 
 		$result = $this->ensureColumnIsNullable($schema, 'share', 'password_by_talk');
 		$result = $this->ensureColumnIsNullable($schema, 'share', 'hide_download') || $result;
-//		$result = $this->ensureColumnIsNullable($schema, 'credentials', 'user') || $result;
```
Comment (nickvergessen) [inline core/Migrations/Version20000Date20201109081915.php:46]: Keep

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: core/Migrations/Version13000Date20170718121200.php (hunk 1 above)
Comment (nickvergessen) [reply core/Migrations/Version13000Date20170718121200.php]: No :(

File: tests/lib/Encryption/ManagerTest.php [hunk 9]
```diff
@@ -193,64 +193,6 @@ public function testSetDefaultEncryptionModule() {
 		$this->assertEquals('ID1', $this->manager->getDefaultEncryptionModuleId());
 	}
 
-//	/**
```
Comment (nickvergessen) [inline tests/lib/Encryption/ManagerTest.php:196]: SHould not be deleted? Someone should reenable and then fix the test instead? :P

File: tests/lib/Files/FilesystemTest.php [hunk 10]
```diff
@@ -325,14 +325,11 @@ public function testHooks() {
 		$rootView->mkdir('/' . $user);
 		$rootView->mkdir('/' . $user . '/files');
 
-//		\OC\Files\Filesystem::file_put_contents('/foo', 'foo');
```
Comment (nickvergessen) [inline tests/lib/Files/FilesystemTest.php:328]: I would keep/fix those. They seem to hint some debugging steps in case the test doesn't work anymore

**Review by nickvergessen — APPROVED** [review]

**Review by come-nc — APPROVED** [review]

**Comment by come-nc** [comment]
@skjnldsv You’ve put milestone 29 for this on purpose? Does it mean we should wait for 28 branch off before merging?

**Comment by nickvergessen** [comment]
> Does it mean we should wait for 28 branch off before merging?

The current idea is to do and merge it directly before the branch-off to break as few PRs as possible and to be still able to somewhat normally use the backport bot to stable28 then

File: vendor-bin/cs-fixer/composer.json [hunk 11]
```diff
@@ -1,11 +1,11 @@
 {
 	"config": {
 		"platform": {
-			"php": "7.4"
+			"php": "8.0"
```
Comment (come-nc) [inline vendor-bin/cs-fixer/composer.json:4]:
```suggestion
			"php": "8.1"
```

**Review by come-nc — CHANGES_REQUESTED** [review]

File: lib/composer/bamarni/composer-bin-plugin/.github/workflows/phpstan.yml
Comment (nickvergessen) [inline lib/composer/bamarni/composer-bin-plugin/.github/workflows/phpstan.yml:1]: Install with `--no-dev` ?

File: vendor-bin/cs-fixer/composer.json [hunk 12]
```diff
@@ -1,11 +1,11 @@
 {
 	"config": {
 		"platform": {
-			"php": "7.4"
+			"php": "8.1"
```
Comment (nickvergessen) [inline vendor-bin/cs-fixer/composer.json:4]: Currently it seems like 28 has to continue to support 8.0.2+

File: lib/composer/bamarni/composer-bin-plugin/.github/workflows/phpstan.yml
Comment (Altahrim) [reply lib/composer/bamarni/composer-bin-plugin/.github/workflows/phpstan.yml:1]: Well spotted!
They were not supposed to be there :)

**Comment by nickvergessen** [comment]
Now is the time

**Comment by nickvergessen** [comment]
rebasing with dropping the `cs:fix` and running it afterwards again

**Review by blizzz — APPROVED** [review]

**Pull request merged**